packages/core/src/telemetry/metrics.ts

```diff
@@ -6,38 +6,305 @@
 import { diag, metrics, ValueType } from '@opentelemetry/api';
-import {
-  SERVICE_NAME,
-  METRIC_TOOL_CALL_COUNT,
-  METRIC_TOOL_CALL_LATENCY,
-  METRIC_API_REQUEST_COUNT,
-  METRIC_API_REQUEST_LATENCY,
-  METRIC_TOKEN_USAGE,
-  METRIC_SESSION_COUNT,
-  METRIC_FILE_OPERATION_COUNT,
-  EVENT_CHAT_COMPRESSION,
-  METRIC_INVALID_CHUNK_COUNT,
-  METRIC_CONTENT_RETRY_COUNT,
-  METRIC_CONTENT_RETRY_FAILURE_COUNT,
-  METRIC_MODEL_ROUTING_LATENCY,
-  METRIC_MODEL_ROUTING_FAILURE_COUNT,
-  METRIC_MODEL_SLASH_COMMAND_CALL_COUNT,
-  // Performance Monitoring Metrics
-  METRIC_STARTUP_TIME,
-  METRIC_MEMORY_USAGE,
-  METRIC_CPU_USAGE,
-  METRIC_TOOL_QUEUE_DEPTH,
-  METRIC_TOOL_EXECUTION_BREAKDOWN,
-  METRIC_TOKEN_EFFICIENCY,
-  METRIC_API_REQUEST_BREAKDOWN,
-  METRIC_PERFORMANCE_SCORE,
-  METRIC_REGRESSION_DETECTION,
-  METRIC_REGRESSION_PERCENTAGE_CHANGE,
-  METRIC_BASELINE_COMPARISON,
-} from './constants.js';
+import { SERVICE_NAME, EVENT_CHAT_COMPRESSION } from './constants.js';
 import type { Config } from '../config/config.js';
 import type { ModelRoutingEvent, ModelSlashCommandEvent } from './types.js';
 
+const TOOL_CALL_COUNT = 'gemini_cli.tool.call.count';
+const TOOL_CALL_LATENCY = 'gemini_cli.tool.call.latency';
+const API_REQUEST_COUNT = 'gemini_cli.api.request.count';
+const API_REQUEST_LATENCY = 'gemini_cli.api.request.latency';
+const TOKEN_USAGE = 'gemini_cli.token.usage';
+const SESSION_COUNT = 'gemini_cli.session.count';
+const FILE_OPERATION_COUNT = 'gemini_cli.file.operation.count';
+const INVALID_CHUNK_COUNT = 'gemini_cli.chat.invalid_chunk.count';
+const CONTENT_RETRY_COUNT = 'gemini_cli.chat.content_retry.count';
+const CONTENT_RETRY_FAILURE_COUNT =
+  'gemini_cli.chat.content_retry_failure.count';
+const MODEL_ROUTING_LATENCY = 'gemini_cli.model_routing.latency';
+const MODEL_ROUTING_FAILURE_COUNT = 'gemini_cli.model_routing.failure.count';
+const MODEL_SLASH_COMMAND_CALL_COUNT =
+  'gemini_cli.slash_command.model.call_count';
+
+// Performance Monitoring Metrics
+const STARTUP_TIME = 'gemini_cli.startup.duration';
+const MEMORY_USAGE = 'gemini_cli.memory.usage';
+const CPU_USAGE = 'gemini_cli.cpu.usage';
+const TOOL_QUEUE_DEPTH = 'gemini_cli.tool.queue.depth';
+const TOOL_EXECUTION_BREAKDOWN = 'gemini_cli.tool.execution.breakdown';
+const TOKEN_EFFICIENCY = 'gemini_cli.token.efficiency';
+const API_REQUEST_BREAKDOWN = 'gemini_cli.api.request.breakdown';
+const PERFORMANCE_SCORE = 'gemini_cli.performance.score';
+const REGRESSION_DETECTION = 'gemini_cli.performance.regression';
+const REGRESSION_PERCENTAGE_CHANGE =
+  'gemini_cli.performance.regression.percentage_change';
+const BASELINE_COMPARISON = 'gemini_cli.performance.baseline.comparison';
+
+const baseMetricDefinition = {
+  getCommonAttributes: (config: Config): Attributes => ({
+    'session.id': config.getSessionId(),
+  }),
+};
+
+const COUNTER_DEFINITIONS = {
+  [TOOL_CALL_COUNT]: {
+    description: 'Counts tool calls, tagged by function name and success.',
```

Comment:
As I read on down the CL here, I definitely feel like an interface that defines each of these fields (maybe with some nice docblocks that describe each field).  I've nearly read this entire CL, and I still don't understand what the `assign` does.  I think an interface with a docblock would make that more obvious to the reader.